code/HabitTracker/app/src/main/java/com/example/habittracker/activities/eventlist/MapsActivity.java

```diff
@@ -0,0 +1,210 @@
+        if (ContextCompat.checkSelfPermission(this,
+                Manifest.permission.ACCESS_FINE_LOCATION)
+                == PackageManager.PERMISSION_GRANTED) {
+            mMap.setMyLocationEnabled(true);
+        } else {
+            ActivityCompat.requestPermissions(this, new String[]
+                            {Manifest.permission.ACCESS_FINE_LOCATION},
+                    REQUEST_LOCATION_PERMISSION);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+        // Check if location permissions are granted and if so enable the
+        // location data layer.
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        switch (requestCode) {
+            case REQUEST_LOCATION_PERMISSION:
+                if (grantResults.length > 0
+                        && grantResults[0]
+                        == PackageManager.PERMISSION_GRANTED) {
+                    enableMyLocation();
+                    break;
+                }
+        }
+    }
+
+    private void setMapLongClick(final GoogleMap map) {
+
+        map.setOnMapLongClickListener(new GoogleMap.OnMapLongClickListener() {
+            @Override
+            public void onMapLongClick(LatLng latLng) {
+                String snippet = String.format(Locale.getDefault(),
+                        "Lat: %1$.5f, Long: %2$.5f",
+                        latLng.latitude,
+                        latLng.longitude);
+
+                Geocoder geocoder;
+                List<Address> addresses;
+                geocoder = new Geocoder(MapsActivity.this, Locale.getDefault());
+                try {
+                    addresses = geocoder.getFromLocation(latLng.latitude, latLng.longitude, 1);
+                    String fullAddress = addresses.get(0).getAddressLine(0);
+                    myAddress = fullAddress;
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+
+                Toast.makeText(MapsActivity.this, myAddress, Toast.LENGTH_LONG).show();
+                myLatitude = latLng.latitude;
+                myLongitude = latLng.longitude;
+                map.addMarker(new MarkerOptions().position(latLng)
+                        .title(myAddress));
+            }
+        });
+    }
+
+    private void setPoiClick(final GoogleMap map) {
+        map.setOnPoiClickListener(new GoogleMap.OnPoiClickListener() {
+            @Override
+            public void onPoiClick(PointOfInterest poi) {
+                Marker poiMarker = mMap.addMarker(new MarkerOptions()
+                        .position(poi.latLng)
+                        .title(poi.name));
+                poiMarker.showInfoWindow();
+                poiMarker.setTag("poi");
+            }
+
+        });
+    }
+
+
+
+
+
+//    @Override
```

Comment:
If we're not using this, it probably shouldn't be in our merged code

Reply:
That's Aparna's section, I'm not sure if she still need it.

Reply:
@coconutwaters

Reply:
I saved a copy and removed those comments